### ISSUE TITLE
[SCREENREADER]: GIBCT EYB - iOS VoiceOver MUST read out the aria-label of the learn more buttons

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#10201]
* [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

**Feedback framework**

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

In Staging Review testing, I discovered that iOS VoiceOver does not read the rich aria-labels for our learn more buttons aloud when they are wrapped in a label for inputs or select menus. I've pinged the tools team about [#4430 - Update the Additional Info component to be used for Learn More expandable sections in forms](https://github.com/department-of-veterans-affairs/va.gov-team/issues/4330) as a blocking item. Screenshot attached.

## Related Issues
* https://github.com/department-of-veterans-affairs/va.gov-team/issues/4330

## Point of Contact

**VFS Point of Contact:** _Trevor_

## Acceptance Criteria

- [ ] iOS Safari reads out the full aria-label correctly

## Environment

* iOS 13.5.1 iPhone 8
* Safari
* VoiceOver
* https://staging.va.gov/gi-bill-comparison-tool/profile/11806124

## Steps to Recreate

1. Start iOS VoiceOver
1. Swipe right through the page until you get to the first Learn more button (looks like a link) that should read out as "Learn more about VA education and training programs".
1. Verify the button is not read aloud

## WCAG or Vendor Guidance (optional)

* [Info and Relationships: Understanding SC 1.3.1](https://www.w3.org/TR/UNDERSTANDING-WCAG20/content-structure-separation-programmatic.html)

## Screenshots or Trace Logs

![Screen Shot 2020-06-12 at 4 19 48 PM](https://user-images.githubusercontent.com/934879/84547233-be0a3c80-acc8-11ea-970a-4d2f1d0dfa53.png)

**Comment by desireeturner** [comment]
This issue is blocked by Update the Additional Info component to be used for Learn More expandable sections in forms #4330. Moving to the Blocked column until #4330 is resolved.

**Comment by desireeturner** [comment]
#4330 is closed now so moving this back to Current Sprint to be picked back up.

**Comment by ksantiagomorgan6** [comment]
Can not test this issue without a local build on an iPhone, The xcode accessibility inspector shows the same for working and non working learn more links.

**Comment by 1Copenut** [comment]
@ksantiagomorgan6 You should be able to test locally with an iDevice using the following command while `vets-website` is running:

> `$ yarn watch --host 0.0.0.0 --public 192.168.x.x:3001`
> Note that we use CORS to limit what hosts can access different APIs, so accessing with a 192.168.x.x address may run into problems

I've used this a fair bit to do local iPhone testing. LMK if you have questions or issues.

**Comment by elder-joseph** [comment]
Associated PR: https://github.com/department-of-veterans-affairs/vets-website/pull/14152

**Comment by sonntag-adam** [comment]
@1Copenut this is in staging.

**Comment by 1Copenut** [comment]
@sonntag-adam 
I reviewed this in staging briefly. The change causes a break in expected `label` behavior by moving the label into the button that triggers the modal. Ideally, the label is associated with the input, and the "Learn more" button is outside the label, as a standalone element. Please review this codepen I created ( https://codepen.io/tpierce_402/full/PoYVKKa ) for a way to mark up the **Learn more** button outside the label and retain the semantic relationship between label + input. The School Information block is the best representative example. It will also require some JS to handle the SPACE keypress -- I had to code the Learn more as a `link[role="button"]` to smooth out screen reader inconsistencies.

---
<img width="509" alt="Screen Shot 2020-09-16 at 12 30 43 PM" src="https://user-images.githubusercontent.com/934879/93371833-99b68280-f818-11ea-989d-7ef465aae596.png">

**Comment by desireeturner** [comment]
@elder-joseph See feedback above.

**Comment by 1Copenut** [comment]
@desireeturner Just verified this issue is fixed. Moving to close.
